### PR TITLE
disk/net/pmem unplug

### PR DESCRIPTION
**Pull request by rbradford**
Support unplugging the same types of virtio-pci devices that we support for hotplug (disk, net and pmem)

### CODE REVIEW

File: vmm/src/device_manager.rs [hunk 1]
```diff
@@ -1712,8 +1735,8 @@ impl DeviceManager {
             id.clone()
         } else {
             let id = self.next_device_name(VFIO_DEVICE_NAME_PREFIX)?;
-            device_cfg.id = Some(id.clone());
-            id
+            device_cfg.id = id.clone();
+            id.unwrap()
```
Comment (sboeuf) [inline vmm/src/device_manager.rs:1738]: unwrap()?

**Comment by sboeuf** [comment]
@sameo PTAL

File: vmm/src/device_manager.rs (hunk 1 above)
Comment (rbradford) [reply vmm/src/device_manager.rs:1738]: @sboeuf Yes the previously the function returned a `String` which was then wrapped into a `Some` and saved with the plain `String` returned. Now it returns an `Option<String>` which is saved directly but the `String` needs to be returned so it needs to be unwrapped.

Comment (rbradford) [reply vmm/src/device_manager.rs:1738]: We don't have to worry about `None` here as on the `feature = "pci_support" case the function either errors or returns `Some(..)`

**Pull request merged**